azure/table_azure_kubernetes_service_version.go

```diff
@@ -0,0 +1,167 @@
+			},
+			{
+				Name:        "type",
+				Description: "Type of the orchestrator version profile list result.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "orchestrator_type",
+				Description: "The orchestrator type.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "orchestrator_version",
+				Description: "Orchestrator version (major, minor, patch).",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "default",
+				Description: "Installed by default if version is not specified.",
+				Type:        proto.ColumnType_BOOL,
+			},
+			{
+				Name:        "is_preview",
+				Description: "Whether Kubernetes version is currently in preview.",
+				Type:        proto.ColumnType_BOOL,
+			},
+			{
+				Name:        "resource_type",
+				Description: "Whether Kubernetes version is currently in preview.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromQual("resource_type"),
+			},
+			{
+				Name:        "upgrades",
+				Description: "The list of available upgrade versions.",
+				Type:        proto.ColumnType_JSON,
+			},
+
+			// Steampipe standard columns
+			{
+				Name:        "title",
+				Description: ColumnDescriptionTitle,
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("Name"),
+			},
+			{
+				Name:        "akas",
+				Description: ColumnDescriptionAkas,
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.FromField("ID").Transform(idToAkas),
+			},
+
+			// Azure standard columns
+			{
+				Name:        "location",
+				Description: ColumnDescriptionRegion,
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromQual("location"),
+			},
+		}),
+	}
+}
+
+type OrchestratorInfo struct {
+	// ID - READ-ONLY; Id of the orchestrator version profile list result.
+	ID *string `json:"id,omitempty"`
+	// Name - READ-ONLY; Name of the orchestrator version profile list result.
+	Name *string `json:"name,omitempty"`
+	// Type - READ-ONLY; Type of the orchestrator version profile list result.
+	Type *string `json:"type,omitempty"`
+	// OrchestratorType - Orchestrator type.
+	OrchestratorType *string `json:"orchestratorType,omitempty"`
+	// OrchestratorVersion - Orchestrator version (major, minor, patch).
+	OrchestratorVersion *string `json:"orchestratorVersion,omitempty"`
+	// Default - Installed by default if version is not specified.
+	Default *bool `json:"default,omitempty"`
+	// IsPreview - Whether Kubernetes version is currently in preview.
+	IsPreview *bool `json:"isPreview,omitempty"`
+	// Upgrades - The list of available upgrade versions.
+	Upgrades *[]containerservice.OrchestratorProfile `json:"upgrades,omitempty"`
+}
+
+//// LIST FUNCTION
+
+func listAKSOrchestractors(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+	location := d.EqualsQualString("location")
+	resourceType := d.EqualsQualString("resource_type")
```

Comment:
Should we not check if `location` and `resourceType` parameters are empty or not?

Reply:
@misraved, Thank you for catching that.

- Looking at the list config, the `location` is required in the WHERE clause for querying this table.
```
KeyColumns: plugin.KeyColumnSlice{
				{
					Name:    "location",
					Require: plugin.Required,
				},
				{
					Name:    "resource_type",
					Require: plugin.Optional,
				},
			},
```
- If we do not provide the `location` in the WHERE clause, Steampipe will throw an error stating that the `location` is required for querying this table.
- On the other hand, the `resource_type` is optional in the WHERE clause, meaning we can choose to include it or not.
- If we include the `resource_type` in the WHERE clause, the API will filter the response based on the specified value. Otherwise, it will return all resource types.

I don't believe we need to perform any empty checks in this case. Thanks again!